doc/source/whatsnew/v0.24.0.rst

```diff
@@ -1334,6 +1334,8 @@ MultiIndex
 I/O
 ^^^
 
+- Bug in :meth:`DataFrame.to_dict` when the resulting dict contains non-Python scalars in the case of numeric data (:issue:`23753`)
+
```

Comment:
Move this under the mini-section.

Reply:
@gfyoung what do you mean by the mini-section?

Reply:
Move your `whatsnew` note under the mini-section titled "Proper handling of np.NaN in a string data-typed column with the Python engine"